plugins/headlamp-plugin/bin/headlamp-plugin.js

```diff
@@ -586,6 +609,44 @@ function getNpmOutdated() {
   return JSON.parse(result);
 }
 
+/**
+ * Use fetch to get the latest package version from the npm registry.
+ *
+ * @returns {Promise<Object>} - promise that resolves to an object containing version information
+ * if the package is outdated, or an empty object.
+ */
+async function getLatestPkgVersion() {
```

Comment:
You introduced a new way to check for the latest version. But the previous function is still used for the upgrade. Should we unify this?